### ISSUE TITLE
Provide a builder like api for binary benchmarks in addition to the macro api

### ISSUE THREAD

**Issue by Joining7943** [Joining7943/iai-callgrind#3]
This issue came up in #2

Binary benchmarks can be currently setup with the `main` macro. The macro api is short and concise but macros are a little bit uncomfortable to work with, so a builder like api could provide a more convenient but also more verbose method to setup binary benchmarks.

**Comment by Joining7943** [comment]
Hi @epage. Have you tried the new version `0.5.0` of iai-callgrind with #2 merged? Are you still interested in using iai-callgrind and having a builder like api?

**Comment by epage** [comment]
I am still interested in having a solution for this need and consider a criterion-like API a hard requirement.  It doesn't have to be a 1:1 and doesn't have to all exist from the beginning but putting everything in a single macro and trying to express the relevant needs is not a sustainable solution for adoption in my mind.

**Comment by Joining7943** [comment]
Ok, great! So, I haven't lost you on the way. Sorry, that it took some time but I found this change was too big to do it in a single step. I hope you find what you were looking for.

I finished the builder like api (so far) and it provides the same functionality (and more) of the macro api. You'll see the new builder api has some similarities to the criterion api. Everything's documented in the README and in the library documentation. If you're missing something, feel free to comment or open an issue.
